include/fmt/std.h

```diff
@@ -35,9 +35,9 @@
 #    if FMT_HAS_INCLUDE(<optional>)
 #      include <optional>
 #    endif
-#    if FMT_HAS_INCLUDE(<source_location>)
-#      include <source_location>
-#    endif
+#  endif
+#  if FMT_CPLUSPLUS > 201703L && FMT_HAS_INCLUDE(<source_location>)
```

Comment:
``<ranges>`` was implemented in compilers before the full implementation of the C++20 standard.
We will need to select the minimum ``__cplusplus`` value from all compilers.
Additionally, the ``>`` is used to check for ``<expected>``.

I think it's better to use ``>`` to indicate that we need a standard newer than ``C++NNN`` for compatible with various compilers.

Reply:
Makes sense but I suggest adding a short comment to clarify why > is intentional here.

Reply:
@vitaut 
Please, help me write a short comment in English.

Reply:
Something along the lines of 

> Use > instead of >= in the version check because source_location may be available after C++17 but before C++20 is marked as implemented.

Reply:
@vitaut 
Done